src/rpdk/core/contract/resource_generator.py

```diff
@@ -247,7 +256,9 @@ def generate_string_strategy(schema):
             if "maxLength" in schema:  # pragma: no cover
                 LOG.warning("found maxLength used with pattern")
 
-            return from_regex(terminate_regex(regex))
+            return from_regex(
+                re.compile(terminate_regex(regex), re.ASCII), fullmatch=True
```

Comment:
terminate_regex replaces `$` with `\Z`... I wonder if should also replace `^` with `\A` then we would drop full match?

```
If you want the generated string to match the whole regex you should use boundary markers.
So e.g. r"\A.\Z" will return a single character string, while "." will return any string, and r"\A.$" 
will return a single character optionally followed by a "\n". Alternatively, passing 
fullmatch=True will ensure that the whole string is a match, as if you had used 
the \A and \Z markers.
```